app/grandchallenge/algorithms/serializers.py

```diff
@@ -7,19 +7,22 @@
     StringRelatedField,
 )
 
-from grandchallenge.algorithms.models import (
-    Algorithm,
-    AlgorithmImage,
-    Job,
+from grandchallenge.algorithms.models import Algorithm, AlgorithmImage, Job
+from grandchallenge.components.models import (
+    ComponentInterface,
+    ComponentInterfaceValue,
 )
 from grandchallenge.components.serializers import (
+    ComponentInterfaceSerializer,
+    ComponentInterfaceValuePostSerializer,
     ComponentInterfaceValueSerializer,
     HyperlinkedComponentInterfaceValueSerializer,
 )
 
 
 class AlgorithmSerializer(serializers.ModelSerializer):
     average_duration = SerializerMethodField()
+    inputs = ComponentInterfaceSerializer(many=True)
```

Comment:
It would be useful to have the outputs here too, make sure both are prefetched in the views.